test/transportSpec.js

```diff
@@ -148,69 +158,99 @@ describe('Transport', function () {
     })
 
-    it('should not accept transfer without condition', function * () {
+    it('should ignore transfer without condition', function * () {
       delete this.params.transfer.executionCondition
-      yield expect(Transport._validateTransfer(this.params))
-        .to.be.rejectedWith(/no-execution/)
+      assert.equal(
+        yield Transport._validateOrRejectTransfer(this.params),
+        'no-execution')
     })
 
     it('should not accept transfer for other account', function * () {
-      this.params.transfer.data.ilp_header.address = 'test.example.garbage'
-      yield expect(Transport._validateTransfer(this.params))
-        .to.be.rejectedWith(/not-my-packet/)
+      this.params.transfer.data.ilp_header.account =
+        'test.example.garbage'
+      assert.equal(
+        yield Transport._validateOrRejectTransfer(this.params),
+        'not-my-packet')
     })
 
     it('should not accept transfer for other protocol', function * () {
-      this.params.transfer.data.ilp_header.address = 'test.example.alice.~ekp'
-      yield expect(Transport._validateTransfer(this.params))
-        .to.be.rejectedWith(/not-my-packet/)
+      this.params.transfer.data.ilp_header.account =
+        'test.example.alice.~ekp'
+      assert.equal(
+        yield Transport._validateOrRejectTransfer(this.params),
+        'not-my-packet')
     })
 
     it('should not accept transfer for other receiver', function * () {
-      this.params.transfer.data.ilp_header.address =
+      this.params.transfer.data.ilp_header.account =
         'test.example.alice.~ipr.garbage'
 
-      yield expect(Transport._validateTransfer(this.params))
-        .to.be.rejectedWith(/not-my-packet/)
+      assert.equal(
+        yield Transport._validateOrRejectTransfer(this.params),
+        'not-my-packet')
     })
 
-    it('should not accept transfer for too little money', function * () {
+    it('should reject transfer for too little money', function * () {
       this.params.transfer.amount = '0.1'
-      yield expect(Transport._validateTransfer(this.params))
-        .to.be.rejectedWith(/insufficient/)
+      assert.equal(
+        yield Transport._validateOrRejectTransfer(this.params),
+        'insufficient')
+      yield this.rejected
     })
 
-    it('should not accept transfer for too much money', function * () {
+    it('should reject transfer for too much money', function * () {
       this.params.transfer.amount = '1.1'
-      yield expect(Transport._validateTransfer(this.params))
-        .to.be.rejectedWith(/overpayment/)
+      assert.equal(
+        yield Transport._validateOrRejectTransfer(this.params),
+        'overpayment')
+      yield this.rejected
     })
 
     it('should accept extra money with "allowOverPayment"', function * () {
       this.params.transfer.amount = '1.1'
       this.params.allowOverPayment = true
-      yield Transport._validateTransfer(this.params)
+      // no error-code is returned on success
```

Comment:
autoFulfillCondition checks truthyness to make sure an error code isn't being returned